overlord/snapstate/autorefresh.go

```diff
@@ -139,37 +142,25 @@ func (m *autoRefresh) LastRefresh() (time.Time, error) {
 }
 
 // EffectiveRefreshHold returns the time until to which refreshes are
-// held if refresh.hold configuration is set and accounting for the
-// max postponement since the last refresh.
+// held if refresh.hold configuration is set.
 func (m *autoRefresh) EffectiveRefreshHold() (time.Time, error) {
-	var holdTime time.Time
+	var holdValue string
 
 	tr := config.NewTransaction(m.state)
-	err := tr.Get("core", "refresh.hold", &holdTime)
+	err := tr.Get("core", "refresh.hold", &holdValue)
 	if err != nil && !config.IsNoOption(err) {
 		return time.Time{}, err
 	}
 
-	// cannot hold beyond last-refresh + max-postponement
-	lastRefresh, err := m.LastRefresh()
-	if err != nil {
-		return time.Time{}, err
+	if string(holdValue) == "forever" {
+		return timeNow().Add(maxDuration), nil
 	}
-	if lastRefresh.IsZero() {
-		seedTime, err := getTime(m.state, "seed-time")
```

Comment:
I see that you removed seed-time for good; is this intended? If so, it would be nice to explain in the commit message why we don't use this anymore.

Reply:
Yes, since we want to allow indefinite holds on all snaps, we no longer need to determine when the last refresh happened to see if the maximum postponement is exceeded.